channels/offline-channel-js/src/OfflineBatchHandler.ts

```diff
@@ -0,0 +1,387 @@
+                            let result = {state: eBatchSendStatus.Complete, data: null} as IOfflineBatchResponse;
+                            //_evts = evts.length;
+                            if (evts.length) {
+                                // make sure doawait has resolve
+                                return doAwaitResponse(provider.removeEvents(evts), (res) => {
+                                    if(res && !res.rejected && res.value) {
+                                        let deletedItems = res.value;
+                                        for (let lp = 0; lp < deletedItems.length; lp++) {
+                                            try {
+                                                
+                                                let evt = evts[lp];
+                                                result.data = evt; // add data
+                                                if (xhrOverride && xhrOverride.sendPOST) {
+                                                    evt.attempCnt++;
+                                                    let sender = xhrOverride.sendPOST; // use transports
+                                                    let onCompleteCallback = (status: number, headers: {
+                                                        [headerName: string]: string;
+                                                    }, res?: string) => {
+                                                     
+                                                        if (status == 200) { // status code (e.g: invalid key) drop payload
+                                                            result.state = eBatchSendStatus.Complete;
+                                                            
+                                                        } else {
+                                                            let isRetriable = _isRetriable(evt, status);
+                                                            if (isRetriable) {
+                                                                result.state = eBatchSendStatus.Retry;
+                                                                _unloadProvider && _unloadProvider.addEvent(evt.id, evt, _itemCtx);
+
+                                                            } else {
+                                                                result.state = eBatchSendStatus.Drop;
+                                                            }
+                                                        }
+                                                        storeResolve(result);
+                                                        
+                                                    };
+                                                    return doAwaitResponse(sender(evt, onCompleteCallback, sync) as any, (res) => {
+                                                        // do nothing, because it should be resolved in callback
```

Comment:
This also looks like it should call the outer resolve / reject as this looks like it would cause the outer promise to never be marked as fullfilled.

Reply:
Oh, I see you have the `storeResult` outer function which calls the `resolve` function